### PR TITLE
Task4 vampires

### PR DESCRIPTION
**Pull request by MariBerdnyk**

### CODE REVIEW

File: GameFighter/Models/Person.cs [hunk 1]
```diff
@@ -12,9 +12,23 @@ public abstract class Person
...
 
-        public virtual void GetAttack(Warrior warrior)
+
+        public virtual int GetAttack(Warrior warrior)
         {
+            int beforeFightHealth = Health;
+            
             Health -= warrior.Attack;
+
+            if (Health > 0)
```
Comment (elkavitaly) [inline GameFighter/Models/Person.cs]: what is the purpose of this part? I'm not sure that you need to handle this case. But if you think that it's okay, you can leave it. In this case use ternary operator

File: GameFighter/Battle.cs [hunk 2]
```diff
@@ -19,13 +19,24 @@ public static bool Fight(Warrior warrior1, Warrior warrior2)
 
             while(warrior1.IsAlive)
             {
-                warrior2.GetAttack(warrior1);
+                int actualattack = warrior2.GetAttack(warrior1);
+
+                if (warrior1 is Vampire)
```
Comment (elkavitaly) [inline GameFighter/Battle.cs]: This logic shouldn't be placed here. Fight class isn't responsible for handling different warrior types.
You need yo carefully use _as_ and _is_ statements. Read this article https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast
The general idea is that when you use these operators they perform casting. This operation requires some resources. In your case there are 2 casings: when you use _is_ operator and _as_. It's not efficient

File: GameFighter/Models/Vampire.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        public int Vampirism { get; protected set; }
+
+        public Vampire()
+        {
+            Health = 40;
+            Attack = 4;
+            Vampirism = 50;
+        }
+
+        public void Healing(int actualAttack)
```
Comment (elkavitaly) [inline GameFighter/Models/Vampire.cs]: This logic shouldn't be in the separate method. Description says that Vampire has this additional healing ability when he attacks

**Review by elkavitaly — CHANGES_REQUESTED** [review]

File: GameFighter/Models/Person.cs (hunk 1 above)
Comment (MariBerdnyk) [reply GameFighter/Models/Person.cs]: I think, that vampires cannot heal from dead fighters

**Review by elkavitaly — APPROVED** [review]

**Pull request merged**